packages/react-native-editor/src/test/index.test.js

```diff
@@ -27,158 +28,218 @@ const initGutenberg = ( registerParams ) => {
+		} );
 	} );
 
-	it( 'dispatches "native.block_editor_props" hook before the editor is rendered', () => {
-		const applyFilters = jest.spyOn( wpHooks, 'applyFilters' );
-
-		// An empty component is provided in order to listen for render calls of the editor component.
-		const onRenderEditor = jest.fn();
-		const EditorComponent = () => {
-			onRenderEditor();
-			return null;
-		};
-		jest.mock( '../setup', () => ( {
-			__esModule: true,
-			default: jest.fn().mockReturnValue( <EditorComponent /> ),
-		} ) );
-
-		initGutenberg();
-
-		const hookCallIndex = 0;
-		// "invocationCallOrder" can be used to compare call orders between different mocks.
-		// Reference: https://git.io/JyBk0
-		const hookCallOrder =
-			applyFilters.mock.invocationCallOrder[ hookCallIndex ];
-		const onRenderEditorCallOrder =
-			onRenderEditor.mock.invocationCallOrder[ 0 ];
-		const hookName = applyFilters.mock.calls[ hookCallIndex ][ 0 ];
-
-		expect( hookName ).toBe( 'native.block_editor_props' );
-		expect( hookCallOrder ).toBeLessThan( onRenderEditorCallOrder );
-	} );
+	describe( 'editor initialization', () => {
+		beforeEach( () => {
+			// Setup already registers blocks so we need assure that no blocks are registered before the test.
+			getBlockTypes().forEach( ( block ) => {
+				unregisterBlockType( block.name );
+			} );
+		} );
 
-	it( 'dispatches "native.render" hook after the editor is rendered', () => {
-		const doAction = jest.spyOn( wpHooks, 'doAction' );
-
-		// An empty component is provided in order to listen for render calls of the editor component.
-		const onRenderEditor = jest.fn();
-		const EditorComponent = () => {
-			onRenderEditor();
-			return null;
-		};
-		jest.mock( '../setup', () => ( {
-			__esModule: true,
-			default: jest.fn().mockReturnValue( <EditorComponent /> ),
-		} ) );
-
-		initGutenberg();
-
-		const hookCallIndex = 1;
-		// "invocationCallOrder" can be used to compare call orders between different mocks.
-		// Reference: https://git.io/JyBk0
-		const hookCallOrder =
-			doAction.mock.invocationCallOrder[ hookCallIndex ];
-		const onRenderEditorCallOrder =
-			onRenderEditor.mock.invocationCallOrder[ 0 ];
-		const hookName = doAction.mock.calls[ hookCallIndex ][ 0 ];
-
-		expect( hookName ).toBe( 'native.render' );
-		expect( hookCallOrder ).toBeGreaterThan( onRenderEditorCallOrder );
-	} );
+		it( 'initializes the editor with empty HTML', async () => {
+			const consoleLog = jest
+				.spyOn( console, 'log' )
+				.mockImplementation( jest.fn() );
+
+			const { getByTestId } = initGutenberg( {}, { initialData: '' } );
+			// Some of the store updates that happen upon editor initialization are executed at the end of the current
+			// Javascript block execution and after the test is finished. In order to prevent "act" warnings due to
+			// this behavior, we wait for the execution block to be finished before acting on the test.
+			await act(
+				() => new Promise( ( resolve ) => setImmediate( resolve ) )
+			);
```

Comment:
If we don't wait for the JS block to be finished, we get a bunch of `act` warnings:
```
Warning: An update to EditorProvider inside a test was not wrapped in act(...).
```

Reply:
Leaving a note that we are actively investigating this issue and approach further in #38052. We should likely retroactively apply whatever global solution we find there to this test.